README.md

```diff
@@ -31,6 +31,130 @@ Application-level tracing for Rust.
 structured, event-based diagnostic information. `tracing` is maintained by the
 Tokio project, but does _not_ require the `tokio` runtime to be used.
 
+## Usage
+
+### In libraries
+
+Libraries should only rely on the `tracing` crate and use the provided macros
+and types to collect whatever information might be useful to downstream consumers.
+
+```toml
+[dependencies]
+tracing = "0.1"
+```
+
+```rust
+use std::{error::Error, io};
+use tracing::{debug, error, info, span, warn, Level, instrument};
+
+#[instrument]
+pub fn shave(yak: usize) -> Result<(), Box<dyn Error + 'static>> {
+    debug!(excitement = "yay!", "hello! I'm gonna shave a yak.");
+    if yak == 3 {
+        warn!("could not locate yak!");
+        // note that this is intended to demonstrate `tracing`'s features, not idiomatic
+        // error handling! in a library or application, you should consider returning
+        // a dedicated `YakError`. libraries like snafu or thiserror make this easy.
+        return Err(io::Error::new(io::ErrorKind::Other, "shaving yak failed!").into());
+    } else {
+        debug!("yak shaved successfully");
+    }
+    Ok(())
+}
+
+pub fn shave_all(yaks: usize) -> usize {
+    let span = span!(Level::TRACE, "shaving_yaks", yaks);
+    let _enter = span.enter();
+
+    info!("shaving yaks");
+
+    let mut yaks_shaved = 0;
+    for yak in 1..=yaks {
+        let res = shave(yak);
+        debug!(yak, shaved = res.is_ok());
+
+        if let Err(ref error) = res {
+            error!(yak, error = error.as_ref(), "failed to shave yak!");
+        } else {
+            yaks_shaved += 1;
+        }
+        debug!(yaks_shaved);
+    }
+
+    yaks_shaved
+}
+```
+
+### In applications
+
+In order to produce output, applications need a concrete subscriber implementation. 
+[`tracing_subscriber`](https://docs.rs/tracing-subscriber/) is a reasonable default.
+By default, `tracing-subscriber` is able to consume messages emitted by
+`log`-instrumented libraries and modules.
+
+```toml
+[dependencies]
+tracing = "0.1"
+tracing-subscriber = "0.2.0-alpha.2"
+```
+
+```rust
+use tracing::{info, Level};
+use tracing_subscriber;
+mod yak_shave;
+
+fn main() {
+    tracing_subscriber::fmt::Subscriber::builder()
+        // all spans/events with a level higher than `DEBUG` (e.g, info, warn, etc.)
+        // will be written to stdout.
+        .with_max_level(Level::DEBUG)
+        // sets this to be the default, global subscriber for this application.
+        .init();
+
+    let number_of_yaks = 3;
+    info!(number_of_yaks, "preparing to shave yaks");
+
+    let number_shaved = yak_shave::shave_all(number_of_yaks);
+    info!(
+        all_yaks_shaved = number_shaved == number_of_yaks,
+        "yak shaving completed."
+    );
+}
+```
+
+### Additional Notes
+
+#### Usage in Asynchronous Code
```

Comment:
Not sure if we really need two section headings...how about just 
```markdown
## Usage

### In Applications

...

### In Libraries

...

### In Asynchronous Code

...

```

Reply:
"In Applications" and "In Libraries" feel like a binary, and asynchronous code is not a fit. That being said, I think this is okay—switching to that.